linker error when json encoding sumtype field where sumtype contains custom type alias.

### Describe the bug

if a struct has a field which is a sumtype, and the sumtype includes a custom user-defined type alias (such as type Mystring = string), there occurs a linker error during causing a compiler bug.

### Expected Behavior

A correctly json encoded struct with empty sumtype field, whatever a sumtype initializes to by default.

### Current Behavior

```
ld: warning: object file (/Users/timurgordon/.vmodules/cache/eb/eb2be82dfdd054420469a9d2d8905481.module.json.o) was built for newer macOS version (13.0) than being linked (10.7)
ld: warning: object file (/Users/timurgordon/code/utils/v/thirdparty/tcc/lib/libgc.a(gc.o)) was built for newer macOS version (12.0) than being linked (10.7)
Undefined symbols for architecture x86_64:
  "_json__decode_main__MyString", referenced from:
      _json__decode_main__MySumType in sumtypemap-b43ed2.o
  "_json__encode_main__MyString", referenced from:
      _json__encode_main__MySumType in sumtypemap-b43ed2.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
builder error: 
==================
C error. This should never happen.
```

### Reproduction Steps

```
module main

import json

struct Test {
	field MySumType
}

type MySumType = string | MyString

type MyString = string

fn main() {
	test := Test{}
	encoded := json.encode(test)
}
```

### Possible Solution

Perhaps fixing the linker warnings, which say `ld: warning: object file (/Users/timurgordon/.vmodules/cache/eb/eb2be82dfdd054420469a9d2d8905481.module.json.o) was built for newer macOS version (13.0) than being linked (10.7)`

### Additional Information/Context

_No response_

### V version

0.3.3

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 f1e9a8f.adcd16b
OS: macos, macOS, 13.2.1, 22D68
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz

--- a/vlib/json/json_encode_primite_test.v
+++ b/vlib/json/json_encode_primite_test.v
@@ -37,5 +37,4 @@ fn test_alias_to_primitive() {
 	mut test3 := MyInt(1000)
 	encoded = json.encode(test3)
 	assert dump(encoded) == '1000'
-
 }

--- a/vlib/json/json_encode_primite_test.v
+++ b/vlib/json/json_encode_primite_test.v
@@ -4,6 +4,7 @@ struct Test {
 	field MySumType
 }
 
+type MyInt = int
 type MyString = string
 type MySumType = MyString | int | string
 
@@ -13,18 +14,21 @@ fn test_alias_to_primitive() {
 	}
 	mut encoded := json.encode(test)
 	assert dump(encoded) == '{"field":"foo"}'
+	assert json.decode(Test, '{"field":	"foo"}')!.field == MySumType('foo')
 
 	test = Test{
 		field: 'foo'
 	}
 	encoded = json.encode(test)
 	assert dump(encoded) == '{"field":"foo"}'
+	assert json.decode(Test, '{"field":"foo"}')! == test
 
 	test = Test{
 		field: 1
 	}
 	encoded = json.encode(test)
 	assert dump(encoded) == '{"field":1}'
+	assert json.decode(Test, '{"field":1}')! == test
 
 	mut test2 := MyString('foo')
 	encoded = json.encode(test2)
@@ -33,4 +37,5 @@ fn test_alias_to_primitive() {
 	mut test3 := MyInt(1000)
 	encoded = json.encode(test3)
 	assert dump(encoded) == '1000'
+
 }

--- a/vlib/json/json_encode_primite_test.v
+++ b/vlib/json/json_encode_primite_test.v
@@ -1,0 +1,36 @@
+import json
+
+struct Test {
+	field MySumType
+}
+
+type MyString = string
+type MySumType = MyString | int | string
+
+fn test_alias_to_primitive() {
+	mut test := Test{
+		field: MyString('foo')
+	}
+	mut encoded := json.encode(test)
+	assert dump(encoded) == '{"field":"foo"}'
+
+	test = Test{
+		field: 'foo'
+	}
+	encoded = json.encode(test)
+	assert dump(encoded) == '{"field":"foo"}'
+
+	test = Test{
+		field: 1
+	}
+	encoded = json.encode(test)
+	assert dump(encoded) == '{"field":1}'
+
+	mut test2 := MyString('foo')
+	encoded = json.encode(test2)
+	assert dump(encoded) == '"foo"'
+
+	mut test3 := MyInt(1000)
+	encoded = json.encode(test3)
+	assert dump(encoded) == '1000'
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -143,9 +143,8 @@ ${enc_fn_dec} {
 			psym := g.table.sym(parent_typ)
 			if is_js_prim(g.typ(parent_typ)) {
 				g.gen_json_for_type(parent_typ)
-				continue
-			}
-			if psym.info is ast.Struct {
+				g.gen_prim_enc_dec(parent_typ, mut enc, mut dec)
+			} else if psym.info is ast.Struct {
 				enc.writeln('\to = cJSON_CreateObject();')
 				g.gen_struct_enc_dec(utyp, psym.info, ret_styp, mut enc, mut dec)
 			} else if psym.kind == .enum_ {
@@ -259,6 +258,16 @@ fn (mut g Gen) gen_enum_enc_dec(utyp ast.Type, sym ast.TypeSymbol, mut enc strin
 			g.gen_enum_to_str(utyp, sym, 'val', 'o', '\t', mut enc)
 		}
 	}
+}
+
+[inline]
+fn (mut g Gen) gen_prim_enc_dec(typ ast.Type, mut enc strings.Builder, mut dec strings.Builder) {
+	type_str := g.typ(typ.clear_flag(.option))
+	encode_name := js_enc_name(type_str)
+	enc.writeln('\to = ${encode_name}(val);')
+
+	dec_name := js_dec_name(type_str)
+	dec.writeln('\tres = ${dec_name}(root);')
 }
 
 [inline]

